.github/actions/setup-kie-tools/action.yml

```diff
@@ -0,0 +1,194 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: "Setup and bootstrap kie-tools repository"
+description: ""
+
+inputs:
+  KOGITO_VERSION:
+    description: "The version for Kogito Runtime"
+    required: false
+    default: "999-SNAPSHOT"
+  DEPS_BRANCHNAME:
+    description: "The branch to pull for Drools, Runtimes and Apps repositories"
+    required: false
+    default: "main"
+  PNPM_FILTER:
+    description: "The filter to use for the bootstrap step"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Get Kiegroup git ref urls
+      env:
+        DROOLS_AND_KOGITO__droolsRepoUrl: https://github.com/kiegroup/drools
+        DROOLS_AND_KOGITO__kogitoRuntimesRepoUrl: https://github.com/kiegroup/kogito-runtimes
+        DROOLS_AND_KOGITO__kogitoAppsRepoUrl: https://github.com/kiegroup/kogito-apps
+      id: get_kiegroup_git_refs
+      shell: bash
+      run: |
+        echo "::group::Get Kiegroup git ref urls"
+
+        # Get Kiegroup Drools Git Ref
+        DROOLS_GIT_REF=$(git ls-remote "${{ env.DROOLS_AND_KOGITO__droolsRepoUrl}}" refs/heads/${{ inputs.DEPS_BRANCHNAME }} | awk '{print $1}')
+        echo "DROOLS_AND_KOGITO__droolsRepoGitRef=$DROOLS_GIT_REF" >> "$GITHUB_OUTPUT"
+
+        # Get Kiegroup Runtimes Git Ref
+        RUNTIMES_GIT_REF=$(git ls-remote "${{ env.DROOLS_AND_KOGITO__kogitoRuntimesRepoUrl}}" refs/heads/${{ inputs.DEPS_BRANCHNAME }} | awk '{print $1}')
+        echo "DROOLS_AND_KOGITO__kogitoRuntimesRepoGitRef=$RUNTIMES_GIT_REF" >> "$GITHUB_OUTPUT"
+
+        # Get Kiegroup Apps Git Ref
+        APPS_GIT_REF=$(git ls-remote "${{ env.DROOLS_AND_KOGITO__kogitoAppsRepoUrl}}" refs/heads/${{ inputs.DEPS_BRANCHNAME }} | awk '{print $1}')
+        echo "DROOLS_AND_KOGITO__kogitoAppsRepoGitRef=$APPS_GIT_REF" >> "$GITHUB_OUTPUT"
+
+        echo "::endgroup::"
+
+    - name: "Cache Maven repository"
+      id: m2
+      uses: actions/cache@v4
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}-drools-${{ steps.get_kiegroup_git_refs.outputs.DROOLS_AND_KOGITO__droolsRepoGitRef}}-runtimes-${{ steps.get_kiegroup_git_refs.outputs.DROOLS_AND_KOGITO__kogitoRuntimesRepoGitRef }}-apps-${{ steps.get_kiegroup_git_refs.outputs.DROOLS_AND_KOGITO__kogitoAppsRepoGitRef }}
+        restore-keys: ${{ runner.os }}-maven
+
+    - name: Clone and Build Drools, Runtimes and Apps repositories
+      if: steps.m2.outputs.cache-hit != 'true'
```

Comment:
With this `if` if a GHA has already built Drools, Runtimes and Apps it will not be built again and will use the cache, skipping the 45min build.

Reply:
Great idea!